cmd/api/insert_test.go

```diff
@@ -0,0 +1,136 @@
+package api
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"scratchdata/config"
+	"scratchdata/models"
+	"scratchdata/pkg/database"
+	memFS "scratchdata/pkg/filestore/memory"
+	memQ "scratchdata/pkg/queue/memory"
+	"scratchdata/pkg/transport/queuestorage"
+)
+
+type testDB struct {
+	database.Database
+}
+
+func (d testDB) Hash(input string) string { return input }
+
+func (d testDB) GetAPIKeyDetails(hashedKey string) models.APIKey {
+	return models.APIKey{
+		ID:           hashedKey,
+		HashedAPIKey: hashedKey,
+	}
+}
+
+func (d testDB) GetDatabaseConnection(connectionID string) models.DatabaseConnection {
+	return models.DatabaseConnection{ID: "test", Type: "memory"}
+}
+
+func TestAPI_Insert(t *testing.T) {
+	cfg := config.API{
+		Enabled:                true,
+		Port:                   0,
+		DataDir:                "",
+		MaxAgeSeconds:          0,
+		MaxSizeBytes:           0,
+		HealthCheckPath:        "",
+		FreeSpaceRequiredBytes: 0,
+	}
+	db := testDB{}
+	qs := queuestorage.NewQueueStorageTransport(queuestorage.QueueStorageParam{
+		Queue:   memQ.NewQueue(),
+		Storage: memFS.NewStorage(),
+		WriterOpt: queuestorage.WriterOptions{
+			DataDir:     t.TempDir(),
+			MaxFileSize: 20,
+			MaxRows:     100,
+			MaxFileAge:  time.Hour,
+		},
+	})
+	api := NewAPIServer(cfg, db, qs)
+
+	const apiKey = "testAPIKey"
+	testCases := []struct {
```

Comment:
These tests should really verify that we are correctly manipulating the input. For example, we should verify that our code has indeed read the table name from the correct location. Or that the output of the `flatten` algorithm is what is expected.